### DISCUSSION TITLE
SQL Query Optimization

### DISCUSSION THREAD

**Discussion by shakerrod** [umami-software/umami#622] [Q&A]
Similar to #592, I have a domain with lots of subdomains used as customer sites. And I use MySQL database.

Therefore I have following questions:
How will the database handle this amount of data? Every customer site will track visits, page views and clicks and there will be millions of records daily.
Will it slow down the real time dashboard, will the queries be slower because of the huge data?

And maybe you have ideas to a less related question about what's got to be done with the database itself to be optimized?
I'm thinking about wiping out the data older than a month ago, collect data within one day and then remove this data and create just reports for days or maybe do some partitioning by day.

Thanks for the help.

**Comment by mikecao** [comment]
Your database will need to scale according to your traffic.  That is getting more into architecture and is outside the scope of the app. The data is indexed though, so having more data shouldn't affect individual website queries. The whole dashboard view might be slow until #592 is resolved.

**Comment by shakerrod** [comment]
@mikecao, thanks for the answer and your work.
